### PR TITLE
(CAT-1226) - Remove Compatibility for Puppet 7.11 and below

### PR DESCRIPTION
**Pull request by jordanbreen28**
## Summary

This PR:
- Removes compatibility with Puppet versions 7.11 and below

## Additional Context

Contains commits from https://github.com/puppetlabs/rspec-puppet/pull/68, thanks to @ekohl.

## Checklist
- [ ] 🟢 Spec tests.
- [ ] 🟢 Acceptance tests.
- [ ] Manually verified.

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
I had hoped you'd take https://github.com/puppetlabs/rspec-puppet/pull/68 as a base.

**Comment by jordanbreen28** [comment]
> I had hoped you'd take #68 as a base.

Apologies... I was actually in the process of commenting on your PR as your comment popped up on this! I was rather silly and hadn't checked for other PRs before commencing the work. Looking to incorporate most of the changes you have in your PR into this one so happy to collaborate

**Comment by ekohl** [comment]
https://github.com/puppetlabs/rspec-puppet/pull/68#discussion_r1299209199 did raise a valid concern with my approach.

I think https://github.com/puppetlabs/rspec-puppet/pull/67 would be better to merge first. I just rebased it and updated it to make RuboCop pass. Let's see if CI agrees.

**Comment by jordanbreen28** [comment]
@ekohl is there any particular reason you opted to setting minimum compatible puppet version to 7.11? We need to support all versions of the two most recent major releases, so will probably need to reduce the minimum supported puppet version to 7.0.

**Comment by bastelfreak** [comment]
Why do you need to support 7.0? That version is dead and won't receive updates anymore? Even the CAT team doesn't support it in most of Puppetlabs modules because it doesn't support the array datatype for exec commands.

**Comment by jordanbreen28** [comment]
@bastelfreak @ekohl I'm going to raise this internally, I'll update the thread when I've more info!

That sounds good to me @ekohl , I'll work to get https://github.com/puppetlabs/rspec-puppet/pull/67 merged and this PR rebased onto it.

**Comment by jordanbreen28** [comment]
@bastelfreak @ekohl thanks for your help with this one, we've decided to go ahead and add compatibility from 7.11.0 onwards.
This is ready for any reviews now.

File: README.md [hunk 1]
```diff
@@ -69,125 +50,83 @@ RSpec.configure do |c|
...
+String | Puppet's default value | any
 
 The path to `puppet.conf`.
 
-#### manifest
-Type   | Default                | Puppet Version(s)
------- | ---------------------- | -----------------
-String | Puppet's default value | 2.x, 3.x
-
-The entry-point manifest for Puppet, usually `$manifest_dir/site.pp`.
```
Comment (nabertrand) [inline README.md:151]: It looks like we decided to keep the `manifest` setting so we should probably keep some sort of documentation. Although I'm not sure this documentation is exactly correct. Currently, it seems like the `manifest` setting is used by rspec-puppet to prepend Puppet code to test:
https://github.com/puppetlabs/rspec-puppet/blob/ef01d40601ee3ed967073108e0cfc58002dd6d5c/lib/rspec-puppet/support.rb#L23-L25
https://github.com/puppetlabs/rspec-puppet/blob/ef01d40601ee3ed967073108e0cfc58002dd6d5c/lib/rspec-puppet/support.rb#L124-L137

File: docs/documentation/configuration/index.md [hunk 2]
```diff
@@ -103,79 +90,28 @@ for further details.
...
-### manifest
-**Type:** String<br />
-**Default:** Puppet's default value<br />
-**Puppet Version(s):** 2.x, 3.x
-
-The entry-point manifest for Puppet, usually `<manifest_dir>/site.pp`.
-
-See the [Puppet
-documentation](https://docs.puppet.com/puppet/latest/configuration.html#manifest)
-for further details.
```
Comment (nabertrand) [inline docs/documentation/configuration/index.md:122]: See above comment about the `manifest` setting

File: README.md (hunk 1 above)
Comment (jordanbreen28) [reply README.md:151]: nice spot @nabertrand - totally missed that.
Will update.

**Review by ekohl — CHANGES_REQUESTED** [review]
I get the impression you took my commit and claimed ownership of it by resetting the author. I'm assuming this is a mistake, rather than intentional but please make sure my original authored code is properly attributed in the git history.

**Comment by jordanbreen28** [comment]
> I get the impression you took my commit and claimed ownership of it by resetting the author. I'm assuming this is a mistake, rather than intentional but please make sure my original authored code is properly attributed in the git history.

Authoring got lost somewhere in the rebasing. I am working to fix this.

**Comment by jordanbreen28** [comment]
@ekohl I hope thats fixed now, apologies!

**Review by GSPatton — APPROVED** [review]

File: lib/rspec-puppet/adapters.rb [hunk 3]
```diff
@@ -141,51 +80,72 @@ def setup_puppet(example_group)
...
-        if (rspec_manifest = RSpec.configuration.manifest)
-          manifest = rspec_manifest
-        else
-          manifest_paths = Puppet[:environmentpath].split(File::PATH_SEPARATOR).map do |path|
-            File.join(path, 'fixtures', 'manifests')
-          end
-
-          manifest = manifest_paths.find do |path|
-            File.exist?(path)
-          end
```
Comment (nabertrand) [inline lib/rspec-puppet/adapters.rb:153]: It seems like this change completely removes use of `manifest` configuration setting. Am I missing where it's still used?

Comment (jordanbreen28) [reply lib/rspec-puppet/adapters.rb:153]: Yes I think you're 100% correct.... Let me update this.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ jobs:
           - ruby-version: '2.7'
             puppet_gem_version: '~> 7.0'
           - ruby_version: '3.2'
-            puppet_gem_version: 'https://github.com/puppetlabs/puppet' # puppet8'
```
Comment (bastelfreak) [inline .github/workflows/ci.yml:21]: Is there a reason for not testing against the branch anymore?

Comment (jordanbreen28) [reply .github/workflows/ci.yml:21]: hey @bastelfreak this was set pre puppet 8 release to allow us to test our tools on puppet 8. We try to test against the latest stable releases, as opposed to main branch as we have seen merged changes cause problems across our CI before.

**Pull request merged**